### PR TITLE
Remove is_theta_argument() function

### PR DESCRIPTION
**Pull request by phate**
This PR does the following:
1. Introduces the ThetaArgument class, which represents a region argument in
a theta subregion.
2. Removes the is_gamma_argument() function.

### CODE REVIEW

File: jlm/rvsdg/theta.hpp [hunk 1]
```diff
@@ -355,6 +355,30 @@ class theta_output final : public structural_output
...
+public:
+  ~ThetaArgument() noexcept override;
+
+private:
+  ThetaArgument(rvsdg::region & region, theta_input & input)
+      : argument(&region, &input, input.Type())
+  {}
+
+  static ThetaArgument &
+  Create(rvsdg::region & region, theta_input & input)
```
Comment (sjalander) [inline jlm/rvsdg/theta.hpp:374]: Is there any way of assuring that the region is a theta region?

**Review by sjalander — DISMISSED** [review]
It would be a bit odd to add a ThetaArgument to any type of region. So, is there a way of checking that the region is part of a Theta operation/node?

File: jlm/rvsdg/theta.hpp (hunk 1 above)
Comment (phate) [reply jlm/rvsdg/theta.hpp:374]: The constructor and creator method of the ThetaArgument class is private and the theta_node class is marked as a friend class. Thus, only the theta_node class is allowed to create a ThetaArgument class. In the theta_node class, the ThetaArgument class is created exactly at one location, and the theta subregion is handed in at that location.

Just to be sure, I added a `JLM_ASSERT()` in the ThetaArgument constructor to check whether it is a theta subregion.

**Comment by phate** [comment]
@sjalander I addressed your comment and need a new review.

**Review by sjalander — APPROVED** [review]

**Pull request merged**